File: Backend/models/pumplog.go
package models

import (
	"time"
)

type PumpLog struct {
	ID          uint    `gorm:"primaryKey"`
	DeviceID    string  `gorm:"type:text;not null"`
	SoilBefore  float64 `gorm:"default:0.0"`
	SoilAfter   float64 `gorm:"default:0.0"`
	TriggeredBy int     `gorm:"not null"` // 1: device, 2: switch, 3: soil, 4: alarm
	StartTime   *time.Time
	EndTime     *time.Time
	CreatedAt   time.Time `gorm:"default:now()"`
	UpdatedAt   time.Time `gorm:"default:now()"`

	Device Device `gorm:"foreignKey:DeviceID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

File: frontend/lib/domain/entities/pumplog.dart
class PumpLogEntity {
  final int? id;
  final int? deviceId;
  final double? soilBefore;
  final double? soilAfter;
  final String? triggeredBy;
  final DateTime? startTime;
  final DateTime? endTime;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const PumpLogEntity({
    this.id,
    this.deviceId,
    this.soilBefore,
    this.soilAfter,
    this.triggeredBy,
    this.startTime,
    this.endTime,
    this.createdAt,
    this.updatedAt,
  });
}
